#543 Add frontend component for the page listing the submissions for a given assessment

kaiyolau opened the pull request:
## Proposed changes

This pull request resolves #543 by adding a page for the GET `/assessments/:assessmentId/submissions` route.

Mentee mode :
![Screen Shot 2023-03-20 at 4 13 37 PM](https://user-images.githubusercontent.com/78048836/226485834-e182186a-0bac-48be-b9c7-62fe230e9dc1.png)

Mentor mode :
![Screen Shot 2023-03-20 at 4 13 47 PM](https://user-images.githubusercontent.com/78048836/226485865-1ce396ec-0f87-4e6c-b9c4-cefd7c8147b0.png)



## Checklist

- [x] Related issue appears at beginning of pull request title with pound sign, and title describes the changes being proposed.
- [x] Related issue is linked in pull request description using [closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).
- [x] The appropriate label has been chosen for this pull request.
- [x] The correct project has been selected for this pull request.
- [x] All commits in this branch, including merge commits, begin with the issue number and a pound sign.
- [x] Tests have been added, where appropriate; or, no tests are relevant for this pull request.
- [x] This pull request contains UI changes, and screenshots of those UI images appear below; or, this pull request contains no UI changes.

kaiyolau committed:
#543 table is working

seidior committed:
#543 Minor tweaks to table appearance

seidior committed:
#543 Add assessment description

seidior commented:
Hi Kyle! Before we can review this, can you please pull the latest from GitHub (I updated your branch with the latest commits from the feature branch), run `yarn delint`, address the comments, and keep running `yarn delint` until you don't get any warnings or errors, then commit and push to GitHub? Thank you!

kaiyolau committed:
#543 addressed the comments

seidior committed:
#543 both participant page and faciltators page finished

seidior commented:
Hi @kaiyolau the force push I just did was to fix the ability to merge this with the current state of the feature branch; this means only your code should now be showing on the PR "Files Changed" page.

seidior reviewed (changes requested):
Great work, Kyle! A couple of modifications I'd like to see, if you don't mind:

- In your PR description, you checked off "This pull request contains UI changes, and screenshots of those UI images appear below; or, this pull request contains no UI changes." but your pull request clearly adds new UI to our webapp, so we'll need a screenshot.
- Can we move the "Mentor Mode" switch above the table (instead of beneath it) and align it to the right side of the container so that it appears above and to the top right of the table?
- In mentor mode, can we show the principal ID of the submissions?
- Finally, can we move the "Take a Quiz!" button to the right side of the container so it appears beneath and to the bottom right of the table? And can we change the text to "New Submission"?

kaiyolau committed:
#543 fix the button to the right and adding principal id

kaiyolau committed:
#543 display button to the right and adding principal ID

kaiyolau commented:
EOD : 
posting a screenshot for the mentor and mentee side to the pull request,
moved the mentor mode switch button to the right,
changed the name of the submission button to "New Submission" and moved it to the right.

seidior commented:
Hi Kyle! Thanks for addressing my feedback. Rather than describing the final tweaks I'd like you to do, it may be more helpful to demonstrate via image:

![submissions_page_facilitator](https://user-images.githubusercontent.com/4006420/226661436-698141f8-a4f7-4b08-8ddd-2fc013fd2b64.png)

Do you mind making the appropriate changes to make this happen? The trickiest part will be the "facilitator mode" switch, which I solved when generating the image above with a nested `<Stack>` inside a `<Grid item>` (but I'm open to other solutions; though don't spend too much time on it, since it's only temporary for now anyway). You can keep the action item for participants as "Review"; only the action items for facilitators should change to the above.

kaiyolau committed:
#543 adding more condition for Action button

kaiyolau commented:
EOD : 

today I have done adding more conditions for the button within the "Action" column, 
and apply "Chip" styling for the "State" column,
also moved the "student ID" column to the first column.

kaiyolau committed:
#543 fix the issue

seidior committed:
#543 Merge latest from feature branch on GitHub

seidior committed:
#543 Minor post-merge-commit tweaks

seidior reviewed (approved):
Hi @kaiyolau! Thank you for incorporating my feedback from my previous review. Everything looks great here! The page renders beautifully, and the outcome here was even better than I'd hoped.

(I did have to temporarily change the due date for the example assessment on my system so that I could see the "New Submission" button, but that's not your fault!)

I had some extra time this morning so I brought your branch up to date with the latest changes from the feature branch so that you're not roadblocked from merger based on that. Once you have two approving reviews, you're welcome to merge!

seidior committed:
#543 Merge latest from feature branch on GitHub

mays4 commented on webapp/src/components/AssessmentSubmissionsListPage.tsx:
```diff
@@ -0,0 +1,250 @@
+  '&:nth-of-type(odd)': {
+    backgroundColor: theme.palette.action.hover,
+  },
+  // hide last border
```
Remove comment

mays4 commented on webapp/src/components/AssessmentSubmissionsListPage.tsx:
```diff
@@ -0,0 +1,250 @@
+  },
+}));
+
+const submissionsExample = [
```
Why you don't move this submissionsExample to data file ?

mays4 reviewed (approved):
Great work , I like the new features added to render based in the role

seidior committed:
#543 Address Mays’ review feedback
